debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeEventSourceCoordinator.java

```diff
@@ -0,0 +1,76 @@
+ */
+package io.debezium.connector.postgresql;
+
+import java.sql.SQLException;
+
+import org.apache.kafka.connect.source.SourceConnector;
+import org.postgresql.replication.LogSequenceNumber;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.connector.postgresql.spi.SlotState;
+import io.debezium.connector.postgresql.spi.Snapshotter;
+import io.debezium.pipeline.ChangeEventSourceCoordinator;
+import io.debezium.pipeline.ErrorHandler;
+import io.debezium.pipeline.EventDispatcher;
+import io.debezium.pipeline.metrics.spi.ChangeEventSourceMetricsFactory;
+import io.debezium.pipeline.source.spi.ChangeEventSource.ChangeEventSourceContext;
+import io.debezium.pipeline.source.spi.ChangeEventSourceFactory;
+import io.debezium.pipeline.source.spi.SnapshotChangeEventSource;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.schema.DatabaseSchema;
+
+public class PostgresChangeEventSourceCoordinator extends ChangeEventSourceCoordinator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(PostgresChangeEventSourceCoordinator.class);
+
+    private final Snapshotter snapshotter;
+    private final SlotState slotInfo;
+
+    public PostgresChangeEventSourceCoordinator(OffsetContext previousOffset, ErrorHandler errorHandler,
+                                                Class<? extends SourceConnector> connectorType,
+                                                CommonConnectorConfig connectorConfig,
+                                                ChangeEventSourceFactory changeEventSourceFactory,
+                                                ChangeEventSourceMetricsFactory changeEventSourceMetricsFactory,
+                                                EventDispatcher<?> eventDispatcher, DatabaseSchema<?> schema,
+                                                Snapshotter snapshotter, SlotState slotInfo) {
+        super(previousOffset, errorHandler, connectorType, connectorConfig, changeEventSourceFactory, changeEventSourceMetricsFactory, eventDispatcher, schema);
+        this.snapshotter = snapshotter;
+        this.slotInfo = slotInfo;
+    }
+
+    @Override
+    protected CatchUpStreamingResult executeCatchUpStreaming(OffsetContext previousOffset, ChangeEventSourceContext context,
+                                                             SnapshotChangeEventSource snapshotSource)
+            throws InterruptedException {
+        if (previousOffset != null && !snapshotter.shouldStreamEventsStartingFromSnapshot() && slotInfo != null) {
+            try {
+                setSnapshotStartLSN((PostgresSnapshotChangeEventSource) snapshotSource,
+                        (PostgresOffsetContext) previousOffset);
+            }
+            catch (SQLException e) {
+                throw new InterruptedException("Failed to determine catch-up streaming stopping LSN");
```

Comment:
`InterruptedException` seems not quite right, it should only be used if indeed a thread was interrupted. Better raise a `DebeziumException`, wrapping the original one.